Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/EditProjectUrlController.java

```diff
@@ -0,0 +1,92 @@
+import edu.asu.spring.quadriga.aspects.annotations.ElementAccessPolicy;
+import edu.asu.spring.quadriga.domain.impl.workbench.Project;
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.workbench.IModifyProjectManager;
+import edu.asu.spring.quadriga.service.workbench.IRetrieveProjectManager;
+import edu.asu.spring.quadriga.validator.ProjectURLValidator;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+@Controller
+public class EditProjectUrlController {
+
+    @Autowired
+    IModifyProjectManager projectManager;
+
+    @Autowired
+    IRetrieveProjectManager retrieveProjectManager;
+
+    @Autowired
+    ProjectURLValidator validator;
+
+    @Resource(name = "projectconstants")
+    private Properties messages;
+
+    private static final Logger logger = LoggerFactory.getLogger(ModifyProjectController.class);
+
+    /**
+     * Attach the custom validator to the Spring context
+     */
+    @InitBinder
+    protected void initBinder(WebDataBinder binder) {
+
+        binder.setValidator(validator);
+    }
+
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/editProjectPageURL/{projectid}", method = RequestMethod.GET)
+    public String editProjectPageURL(@PathVariable("projectid") String projectid, Principal principal, ModelMap model)
+            throws QuadrigaStorageException, QuadrigaAccessException {
+        IProject project = retrieveProjectManager.getProjectDetails(projectid);
+        model.addAttribute("project", project);
+        model.addAttribute("unixnameurl", messages.getProperty("project_unix_name.url"));
+        return "auth/workbench/editProjectPageURL";
+    }
+
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 3, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/editProjectPageURL/{projectid}", method = RequestMethod.POST)
+    public String editProjectPageURL(@Validated @ModelAttribute("project") Project project, BindingResult result,
+            @PathVariable("projectid") String projectid, Principal principal, ModelMap model,
+            RedirectAttributes redirectAttributes) throws QuadrigaStorageException, QuadrigaAccessException {
+        if (result.hasErrors()) {
+            logger.error("Update project details error:", result);
+            model.addAttribute("project", project);
+            model.addAttribute("unixnameurl", messages.getProperty("project_unix_name.url"));
+            return "auth/workbench/editProjectPageURL";
+        } else {
```

Comment:
remove this else. your return in line 82 if there are errors, so no need for this else.